### PR TITLE
MSSQL: decouple plugin

### LINKED ISSUE #89356: MSSQL multi-tenant ready

### LINKED ISSUE #90371: MSSQL: Extract from grafana/grafana
This should be done after [work for the other SQL based data sources](https://github.com/grafana/grafana/issues/72759) have been completed due to dependencies and learnings

### PR DESCRIPTION
**Pull request by bossinc**
Used Gabor's findings on MySQL for this PR.

Gabor's PRs:
https://github.com/grafana/grafana/pull/86168
https://github.com/grafana/grafana/pull/86308
https://github.com/grafana/grafana/pull/87330

### CODE REVIEW

File: pkg/tsdb/mssql/standalone/main.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+	// Start listening to requests sent from Grafana. This call is blocking so
+	// it won't finish until Grafana shuts down the process or the plugin choose
+	// to exit by itself using os.Exit. Manage automatically manages life cycle
+	// of datasource instances. It accepts datasource instance factory as first
+	// argument. This factory will be automatically called on incoming request
+	// from Grafana to create different instances of SampleDatasource (per datasource
+	// ID). When datasource configuration changed Dispose method will be called and
+	// new datasource instance created using NewSampleDatasource factory.
+	logger := backend.NewLoggerWith("logger", "tsdb.mssql")
+	cfg := setting.NewCfg()
```
Comment (bossinc) [inline pkg/tsdb/mssql/standalone/main.go:25]: I am not 100% sure this is good enough.

File: pkg/tsdb/mssql/standalone/main.go [hunk 2]
```diff
@@ -0,0 +1,30 @@
+package main
+
+import (
+	"os"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"github.com/grafana/grafana-plugin-sdk-go/backend/datasource"
+	"github.com/grafana/grafana-plugin-sdk-go/backend/log"
+
+	//"github.com/grafana/grafana/pkg/setting"
```
Comment (bossinc) [inline pkg/tsdb/mssql/standalone/main.go:10]:
```suggestion
```

**Review by asimpson — CHANGES_REQUESTED** [review]
Looks good, just one question from me.

**Review by asimpson — APPROVED** [review]
LGTM :rocket: Just need to get Drone to pass :crossed_fingers:

**Pull request merged**